source/common/router/router.cc

```diff
@@ -859,6 +880,15 @@ void Filter::chargeUpstreamAbort(Http::Code code, bool dropped, UpstreamRequest&
 
 void Filter::onUpstreamTimeoutAbort(StreamInfo::ResponseFlag response_flags,
                                     absl::string_view details) {
+  if (cluster_->timeoutBudgetStats().has_value()) {
+    Event::Dispatcher& dispatcher = callbacks_->dispatcher();
+    std::chrono::milliseconds response_time = std::chrono::duration_cast<std::chrono::milliseconds>(
+        dispatcher.timeSource().monotonicTime() - downstream_request_complete_time_);
+
+    cluster_->timeoutBudgetStats()->upstream_rq_timeout_budget_percent_used_.recordValue(
+        percentageOfTimeout(response_time, timeout_.global_timeout_));
```

Comment:
I may have asked this previously, but in this case don't you know that you know that you used the global_timeout_ so can't you just provide 100 as the used percent?

Reply:
I think I had agreed with you because this was in the path of the global timeout only.  Now this is in the codepath for either exhaustion of per-try requests or the global timeout failing.

In the case where all retries have failed and this is hit, this shouldn't be 100.